issue-008/solution.js

```diff
@@ -1,4 +1,22 @@
 // implement your solution here
-function sum() {
-
-}
\ No newline at end of file
+function sum(a, b, c) {
+    argumentsArray = [];
+    if (arguments.length === 0) {
+        return 0;
+    } else if (arguments.length > 0 && arguments.length <= 3) {
+        sumOfArguments = a + b + c;
+        return sumOfArguments;
+    } else if (arguments.length > 3) {
+        for (var i = 0; i < arguments.length; i++) {
+            argumentsArray.push(arguments[i]);
+        }
+        var arraySum = function(myArray) {
+                var sums = myArray.reduce(add, 0);
```

Comment:
Pius, I think you overusing the `reduce` method. For this, you were just to do a simple loop over the `arguments` object and add up all the numbers:

```javascript
sum = 0;
for (var i = 0; i < arguments.length; i++) {
  sum += arguments[i];
}
return sum;
```

And you don't need the `if`'s that you had up there.